src/Navigation.js

```diff
@@ -48,7 +48,7 @@ function _registerComponentNoRedux(screenID, generator) {
 
       render() {
         return (
-          <InternalComponent testID={screenID} navigator={this.navigator} {...this.state.internalProps} />
+          <InternalComponent testID={screenID} onPress={this.button ? this.button.onPress.bind(this.button) : null} navigator={this.navigator} {...this.state.internalProps} />
```

Comment:
I like the subclass idea...but then it would expose a `registerComponent()` and `registerNavigationButtonComponent()` top-level function as external API. This still seems desirable to me, but I want to confirm before I head down that road.